tests/functional/modules/test_zos_backup_restore.py

```diff
@@ -930,83 +929,126 @@ def test_backup_and_restore_a_data_set_with_same_hlq(ansible_zos_module):
+            volume=volume_2,
+            hlq=hlqs,
+        )
+        assert_module_did_not_fail(results)
+        assert_data_set_exists(hosts, data_set_restore_location)
+    finally:
+        delete_data_set_or_file(hosts, data_set_name)
+        delete_data_set_or_file(hosts, data_set_restore_location)
+        delete_remnants(hosts, hlqs)
 
 
-# def test_backup_and_restore_of_full_volume(ansible_zos_module):
-#     hosts = ansible_zos_module
-#     try:
-#         delete_data_set_or_file(hosts, DATA_SET_BACKUP_LOCATION)
-#         delete_data_set_or_file(hosts, data_set_name)
-#         create_sequential_data_set_with_contents(
-#             hosts, data_set_name, DATA_SET_CONTENTS, VOLUME
-#         )
-#         results = hosts.all.zos_backup_restore(
-#             operation="backup",
-#             volume=VOLUME,
-#             full_volume=True,
-#             sms_storage_class="DB2SMS10",
-#             backup_name=DATA_SET_BACKUP_LOCATION,
-#             overwrite=True,
-#             space=500,
-#             space_type="m",
-#         )
-#         assert_module_did_not_fail(results)
-#         assert_data_set_or_file_exists(hosts, DATA_SET_BACKUP_LOCATION)
-#         delete_data_set_or_file(hosts, data_set_name)
-#         results = hosts.all.zos_backup_restore(
-#             operation="restore",
-#             backup_name=DATA_SET_BACKUP_LOCATION,
-#             overwrite=True,
-#             volume=VOLUME,
-#             full_volume=True,
-#             sms_storage_class="DB2SMS10",
-#             space=500,
-#             space_type="m",
-#         )
-#         assert_module_did_not_fail(results)
-#         assert_data_set_exists_on_volume(hosts, data_set_name, VOLUME)
-#     finally:
-#         delete_data_set_or_file(hosts, data_set_name)
-#         delete_data_set_or_file(hosts, DATA_SET_BACKUP_LOCATION)
+def test_backup_and_restore_of_sms_group(ansible_zos_module, volumes_sms_systems):
+    hosts = ansible_zos_module
+    data_set_name = get_tmp_ds_name()
+    data_set_backup_location = get_tmp_ds_name()
+    try:
+        volumes = Volume_Handler(volumes_sms_systems)
+        volume, smsgrp = volumes.get_available_vol_sms()
+        delete_data_set_or_file(hosts, data_set_backup_location)
+        delete_data_set_or_file(hosts, data_set_name)
+        create_sequential_data_set_with_contents(
+            hosts, data_set_name, DATA_SET_CONTENTS, volume
+        )
+        sms = {"storage_class":smsgrp}
+        results = hosts.all.zos_backup_restore(
+            data_sets=dict(include=data_set_name),
+            operation="backup",
+            volume=volume,
+            backup_name=data_set_backup_location,
+            overwrite=True,
+            sms=sms,
+        )
+        assert_module_did_not_fail(results)
+        assert_data_set_or_file_exists(hosts, data_set_backup_location)
+        delete_data_set_or_file(hosts, data_set_name)
+        sms = {
+            "disable_automatic_class":['data_set_name'],
```

Comment:
This is odd and shouldn't be working, this and the `keyword` wrong keys shows that is definetly something odd with the tests and needs to be checked more carefully, we may be testing in a wrong way. Did you tried running this manually and ensured that the classes were properly assigned? 


```suggestion
            "disable_automatic_class":[data_set_name],
```